docs/developer/pull-requests.md

```diff
@@ -3,6 +3,142 @@ This document outlines best practices to follow when creating and reviewing PRs.
 
 ## As an author
 
+### Pull request process
+1. Create a GitHub PR from your fork
+1. Verify all checks are passing
+1. Get someone to review it
+1. Iterate your changes based on the feedback you receive. Don't rewrite the history (rebase or squash) with your new changes, keep them as individual commits. This facilitates the reviewers job.
```

Comment:
I'm OK with advocating not squashing updates until the end but is it really necessary to mandate no re-writing. I feel like that judgement should be left to developers.

To provide an example: if I re-write a function based on feedback why shouldn't I squash that, its a new version implemented differently.

This comes back to the auto removal of LGTM too. If we're auto removing LGTM we're forcing developers to do squashing via the UI else the LGTM is removed - that's not good.

Reply:
Resolved offline.